internal/elasticsearch/index/template.go

```diff
@@ -298,6 +313,14 @@ func expandTemplate(config interface{}) (models.Template, bool, diag.Diagnostics
 	}
 	templ.Aliases = aliases
 
+	lifecycle, diags := ExpandLifecycle(definedTempl["lifecycle"].(*schema.Set))
+	if diags.HasError() {
+		return templ, false, diags
+	}
+	if lifecycle != nil {
+		templ.Lifecycle = lifecycle
+	}
```

Comment:
The tests are failing when `lifecycle` is not defined in the TF module. I'm pretty sure this should fix it, though it's totally untested. 
```suggestion
	if lc, ok := definedTempl["lifecycle"]; ok {
		lifecycle, diags := ExpandLifecycle(lc.(*schema.Set))
		if diags.HasError() {
			return templ, false, diags
		}
		if lifecycle != nil {
			templ.Lifecycle = lifecycle
		}
	}
```

Reply:
Sorry for mistake. I didn't test newly created index. I fixed it in `ExpandLifecycle` function.